scripts/maintenance-status/maintenance-status.rb

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/ruby
+
+require "optparse"
+require "open3"
+require "active_support/core_ext/hash/conversions"
+
+class ObsHandler
+  def initialize(config = {})
+    @group = config[:group] || "cloud-maintenance"
+  end
+
+  def requests()
+    result = obs_api_call("/search/request?match=review[@by_group='#{@group}'+and+(@state='new')]")
+    requests = result["request"]
+    open_requests = []
+    requests.each do |request|
+      if request["state"]["name"] = "review" && request["id"]
+        id = request["id"].to_i
+        status = request_status?(request["id"])
+        source = request["action"][1]["source"]["project"].split(":").last
+        products = filter_products(request)
+        open_requests.push(
+          {
+            id: id,
+            source: source,
+            status: status,
+            products: products
+          }
+        )
+      end
+    end
+    open_requests
+  end
+
+  def request_status?(id)
+    comments = obs_api_call("/comments/request/#{id}")["comment"] || ""
+    if comments.select{|comment| comment.match(/^SUSE OpenStack Cloud Jenkins run triggerd!.*/)}
+      return :running
+    elsif comments.select{|comment| comment.match(/^SUSE OpenStack Cloud Jenkins run failed!.*/)}
+      return :failure
+    elsif comments.select{|comment| comment.match(/^SUSE OpenStack Cloud Jenkins run succeeded!.*/)}
+      return :success
+    else
+      return :pending
+    end
+  end
+
+  def request_status(id, status, message = "")
+    unless message == ""
+      message = " (#{message})"
+    end
+    if status == :running
+      obs_api_call(
+        "/comments/request/#{id}",
+        "POST",
+        "SUSE OpenStack Cloud Jenkins run triggerd!#{message}"
+      )
+    elsif status == :failure
+      obs_api_call(
+        "/comments/request/#{id}",
+        "POST",
+        "SUSE OpenStack Cloud Jenkins run failed!#{message}"
+      )
+    elsif status == :success
+      obs_api_call(
+        "/comments/request/#{id}",
+        "POST",
+        "SUSE OpenStack Cloud Jenkins run succeeded!#{message}"
+      )
+    else
+      puts "No valid status gived!"
+    end
+  end
+
+  def open_requests()
+    requests = requests()
+    print_rr_info(requests)
+  end
+
+  def unseen_requests()
+    requests = requests()
+    filter_requests(requests, [:pending])
+    print_rr_info(requests)
+  end
+
+  def forcerebuild_requests()
+    requests = requests()
+    filter_requests(requests, [:running, :pending, :failure])
+    print_rr_info(requests)
+  end
+
+  def filter_requests(requests, filter)
+    requests.select! do |request|
+      filter.include?(request[:status])
+    end
+  end
+
+  def filter_products(request)
```

Comment:
An idea for future improvement, if this does turn out to be something that we maintain: rather than encoding these product values here, it would be better if you used an external JSON or YAML configuration file, which people can edit without having to understand what this script (and others like it) does.